main.js

```diff
@@ -1,51 +1,37 @@
-// function([string1, string2],target id,[color1,color2])    
-consoleText(['SH3LL', 'R3V3R53', 'W3B', 'CRYP70', 'P4WN' ,  '@DHBW Mannheim'], 'text',['0414a7','0414a7','0414a7', '0414a7', '0414a7', '0414a7']);
-
-function consoleText(words, id, colors) {
-  if (colors === undefined) colors = ['#fff'];
-  var visible = true;
-  var con = document.getElementById('console');
-  var letterCount = 1;
-  var x = 1;
-  var waiting = false;
-  var target = document.getElementById(id)
-  target.setAttribute('style', 'color:' + colors[0])
-  window.setInterval(function() {
-
-    if (letterCount === 0 && waiting === false) {
-      waiting = true;
-      target.innerHTML = words[0].substring(0, letterCount)
-      window.setTimeout(function() {
-        var usedColor = colors.shift();
-        colors.push(usedColor);
-        var usedWord = words.shift();
-        words.push(usedWord);
-        x = 1;
-        target.setAttribute('style', 'color:' + colors[0])
-        letterCount += x;
-        waiting = false;
-      }, 1000)
-    } else if (letterCount === words[0].length + 1 && waiting === false) {
-      waiting = true;
-      window.setTimeout(function() {
-        x = -1;
-        letterCount += x;
-        waiting = false;
-      }, 1000)
-    } else if (waiting === false) {
-      target.innerHTML = words[0].substring(0, letterCount)
-      letterCount += x;
-    }
-  }, 120)
-  window.setInterval(function() {
-    if (visible === true) {
-      con.className = 'console-underscore hidden'
-      visible = false;
-
-    } else {
-      con.className = 'console-underscore'
-
-      visible = true;
-    }
-  }, 400)
-}
\ No newline at end of file
+'use strict';
+
+function sleep(ms) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+const textArray = ['SH3LL', 'R3V3R53', 'W3B', 'CRYP70', 'P4WN', '@DHBW Mannheim'];
+const elementId = typewriterText;
+const charTime = 100; // ms
+const sleepTime = 2000; // ms
+
+(async function typeWriter() {
+	for (let i = 0; i < textArray.length; i++) {
+
+		elementId.innerHTML = '';
+		let charArray = textArray[i].split('');
+
+		charArray.forEach((letter, j) => 
+			setTimeout(() => (elementId.innerHTML += letter), charTime * j)
+		);
+		
+		// need to wait for all characters to appear, then sleep
+		let totalSleepTime = sleepTime + charArray.length * charTime;
+		await sleep(totalSleepTime); 
+
+		charArray.forEach((letter, j) => {
+				setTimeout(() => {
+					elementId.innerHTML = elementId.innerHTML.slice(0, -1); 
+				}, charTime * j);
+			}
+		);
+
+		await sleep(totalSleepTime);
+	}
+
+	setTimeout(typeWriter, sleepTime);
```

Comment:
Hier wird nochmal um `sleepTime` verzögert, d.h. die Wartezeit zwischen dme letzten und ersten Element von `textArray` ist `2*sleepTime`. Ist das intended? Ansonsten vielleicht `setTimeout(typeWriter, 0)`?

Reply:
nein, ist nicht intended.
Weiß auch nicht, wie ich da genau drauf gekommen bin.
Du kannst die ganze Funktion in `while(true){}` schreiben, dann brauchen wir gar kein `setTimeout()`